lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -110,6 +121,8 @@ def oversee_users(self, departed_relation_id: Optional[int]):
                 config = self._get_config(username)
                 logger.info("Update relation user: %s on %s", config.username, config.database)
                 mongo.update_user(config)
+                logger.info("Updating relation data according to diff")
+                self._diff(event)
```

Comment:
Should the result of this function be handled in some way on MongoDB side (does it use the diff)?

Reply:
I don't think it directly uses the diff, but I think the diff function issues a relation changed event when it does:
`event.relation.data[self.charm.model.app].update({"data": json.dumps(new_data)})`, which triggers the mongodb code to handle the diff, idk if that makes sense or not.